addons/web/static/src/legacy/js/core/dialog.js

```diff
@@ -104,6 +104,10 @@ var Dialog = Widget.extend({
         this.renderFooter = options.renderFooter;
         this.onForceClose = options.onForceClose;
 
+        const exclusions = this._getScrollableBodyExclusions().map(
+            (exclusion) => `:not(${exclusion})`
+        ).join("");
+        this.remainingModalSelector = `.modal[role="dialog"]${exclusions}`;
```

Comment:
I will be annoying but there are things I would improve. Instead of building a complex selector with `:not` that will need to be re-parsed by jQuery just after, I would find the modal elements with the standard selector and filter them afterwards
    1. I am pretty sure it's faster (although not important, even if slower).
    2. If a selector is something like `.hello > .world`, your code will lead to a `:not(.hello > .world)`... which is only valid if using jQuery. As we are trying to not use jQuery anymore, let's practice :wink: 
    3. When converting this to javascript / getting rid of this legacy file in the future, that will be easier
    4. More flexibility for overrides: instead of having to give a list of string selector, the overrides can do what they want

So something like:
```javascript
const modals = $('.modal[role="dialog"]).filter(':visible').filter(this._yourNiceFunctionName);
```
then
```javascript
_yourNiceFunctionName(el) {
     if (i do not want el) {
        return false;
     }
     return this._super(...arguments);
}
```

Reply:
:+1: 
Exactly what I was refering to here https://github.com/odoo/odoo/pull/83258#discussion_r791863921
Thanks for your confirmation